sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -161,6 +163,19 @@ public static boolean isNullOrEmpty(CharSequence charSequence) {
     }
 
 
+    /**
+     * Helper method that returns {@link UserAgentProperties} from properties defined in {@code propertiesFileName}.
+     *
+     * @param propertiesFileName The file name defining the properties.
+     * @return {@link UserAgentProperties}.
+     */
+    public static UserAgentProperties getUserAgentPropertiesFromProperties(String propertiesFileName) {
```

Comment:
We have `getProperties()` method that returns a map. Client libraries should call that method and look up whatever properties they want. `UserAgentProperties` is too specific and will not allow for additional properties to be read from properties file.

Reply:
That's what object used for. If it needs to have more info for user agent, we can simply extend the object. I reverted the getProperties. If we want to put any other info to properties, there is still a choice.